### PR TITLE
Task/g99 sonar fixes

### PR DESCRIPTION
**Pull request by fboucquez**
Hi,

This PR fixes many sonar issues. From here we should keep sonar happy when adding new features. 
This will be simpler when we have our own sonar doing the PR code reviews for us but the details about where and how sonar will be deployed are under discussion. 

Cheers
Fernando

### CODE REVIEW

File: sdk-core/src/main/java/io/nem/core/crypto/ed25519/arithmetic/Ed25519EncodedGroupElement.java [hunk 1]
```diff
@@ -133,6 +133,6 @@ public boolean equals(final Object obj) {
     @Override
     public String toString() {
         return String.format(
-            "x=%s\ny=%s\n", this.getAffineX().toString(), this.getAffineY().toString());
+            "x=%s\ny=%s%n", this.getAffineX().toString(), this.getAffineY().toString());
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/core/crypto/ed25519/arithmetic/Ed25519EncodedGroupElement.java]: Why we change this?

File: sdk-core/src/main/java/io/nem/sdk/model/account/Address.java [hunk 2]
```diff
@@ -37,23 +37,23 @@
 public class Address {
 
     private static final int NUM_CHECKSUM_BYTES = 4;
-    private final String address;
+    private final String plain;
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/account/Address.java]: I'd use 'plainAddress' to be more meaningful

File: sdk-core/src/main/java/io/nem/core/crypto/ed25519/arithmetic/Ed25519EncodedGroupElement.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/core/crypto/ed25519/arithmetic/Ed25519EncodedGroupElement.java]: https://rules.sonarsource.com/java/tag/confusing/RSPEC-3457

Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/core/crypto/ed25519/arithmetic/Ed25519EncodedGroupElement.java]: https://rules.sonarsource.com/java/tag/confusing/RSPEC-3457

File: sdk-core/src/main/java/io/nem/sdk/model/account/Address.java (hunk 2 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/account/Address.java]: ok, I'll rename

File: sdk-core/src/main/java/io/nem/catapult/builders/AccountRestrictionTypeDto.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
 package io.nem.catapult.builders;
 
 import java.io.DataInput;
+import java.util.Arrays;
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/catapult/builders/AccountRestrictionTypeDto.java]: do we need this change?

Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/catapult/builders/AccountRestrictionTypeDto.java]: pushed

**Pull request merged**